lms/djangoapps/teams/tests/test_views.py

```diff
@@ -709,6 +709,30 @@ def test_access_by_username(self, user, status, has_content):
             else:
                 self.assertEqual(membership['count'], 0)
 
+    def test_username_course_filter(self):
```

Comment:
Awesome that you've picked up on `ddt` already. I recommend that you use it in this test case already so that you can test the following cases as well:
* For the `'student_enrolled_both_courses_other_team'` user:
  * test_course_1
  * test_course_2
* For the `'student_enrolled'` user:
  * test_course_1
  * test_course_2
That way, we can also test for courses that the users are not in/don't have teams in.